docs/api.rst

```diff
@@ -297,6 +297,105 @@ December 2016 onward (i.e. for single-tile products), you can use a `filename` p
 
   kw['filename'] = '*_T{}_*'.format(tile)  # products after 2016-12-01
 
+Exclude Products already available offline from download_all
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``SentinelAPI.download_all()`` automatically skips products that are already in the target directory (current directory
+or as specified in ``directory_path``) if their folder name has not changed. In order to exclude products stored under a
+different location and/or folder name, one has to manually filter the OrderedDict returned from the ``query()``
+method before feeding it to ``download_all()``:
+
+.. code-block:: python
+
+  from sentinelsat import SentinelAPI
+  import xml.etree.ElementTree as ET
+  import os
+  import zipfile
+  import tempfile
+  import time
+  import re
+
+  # make a demo folder:
+  demo_folder = os.path.join(tempfile.gettempdir(),"sentinelsat_exclude_products_demo_{}".format(round(time.time())))
+
+  # On (old) windows systems the maximum path length is restricted to 260 characters. Exceeding this limitation might
+  # result in errors (cp. https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx#maxpath).
+  # user "eryksun" on https://stackoverflow.com/questions/36219317/pathname-too-long-to-open/36237176
+  # offers the following solution (+explanations):
+  if os.name == 'nt':
+    demo_folder = u"\\\\?\\" + demo_folder
+
+  os.mkdir(demo_folder)
+  print("Watch the following folder to see what happens during the demo:\n{}".format(demo_folder))
+
+  # set up a demo query including Sentinel 1 and 2 products
+  api = SentinelAPI('user','password')
+  demo_query = api.query(filename = "S2A_OPER_PRD_MSIL1C_PDMC_20161013T075059_R111_V20161012T161812_20161012T161807.SAFE")
+  demo_query.update(api.query(filename = "S1A_WV_OCN__2SSV_20150526T081641_20150526T082418_006090_007E3E_104C.SAFE"))
+
+  # download the products in the demo_query dictionary
+  downloaded = api.download_all(demo_query, directory_path=demo_folder)
+  zip_1 = [i for i in os.listdir(demo_folder) if i.endswith(".zip")]
```

Comment:
use `glob.glob(os.path.join(demo_folder, '*.zip'))` instead of that list comprehension around `os.listdir`. Same a couple of lines down.

Reply:
Thanks a lot for this comment! I can definitly put this to use in the future.